provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -551,6 +548,15 @@ void hsm_client_riot_destroy(HSM_CLIENT_HANDLE handle)
         /* Codes_SRS_HSM_CLIENT_RIOT_07_008: [ hsm_client_riot_destroy shall free the HSM_CLIENT_HANDLE instance. ] */
         free(x509_client->certificate_common_name);
         /* Codes_SRS_HSM_CLIENT_RIOT_07_009: [ hsm_client_riot_destroy shall free all resources allocated in this module. ] */
+        mbedtls_ecp_point_free(&x509_client->ca_root_pub);
+        mbedtls_mpi_free(&x509_client->ca_root_priv);
+
+        mbedtls_ecp_point_free(&x509_client->device_id_pub);
+        mbedtls_mpi_free(&x509_client->device_id_priv);
```

Comment:
Same here. [`mbedtls_mpi_free`](https://tls.mbed.org/api/bignum_8h.html#abffd5d8382cb86c089a1ef2bebb90f9f) takes [`mbedtls_mpi*`](https://tls.mbed.org/api/structmbedtls__mpi.html) and `device_id_priv` eventually resolves to 
```c
typedef struct {
    uint32_t data[BIGLEN];
} bigval_t;
```

Reply:
That is the old style of struct, are you comparing to an updated RIoT submodule or the current hash? Current hash doesn't have the updated mbed crypto structs, otherwise there is something going on with the includes.

Reply:
Please write & test an application/sample using riot with Valgrind to ensure we're not leaking any memory.

Reply:
I valgrinded on what we have currently and there was leaks hence the new destroy helper functions